samples/csharp_dotnetcore/51.cafe-bot/CafeBot.cs

```diff
@@ -0,0 +1,219 @@
+            _dialogs = new DialogSet(_dialogAccessor);
+            _dialogs.Add(new MainDispatcher(services, _onTurnAccessor, userState, conversationState, loggerFactory));
+        }
+
+        /// <summary>
+        /// Gets the <see cref="ConversationState"/> object for the conversation.
+        /// </summary>
+        /// <value>The <see cref="ConversationState"/> object.</value>
+        public ConversationState ConversationState { get; }
+
+        /// <summary>
+        /// Gets the <see cref="UserState"/> object for the conversation.
+        /// </summary>
+        /// <value>The <see cref="UserState"/> object.</value>
+        public UserState UserState { get; }
+
+        /// <summary>
+        /// Every conversation turn for our Basic Bot will call this method.
+        /// </summary>
+        /// <param name="context">A <see cref="ITurnContext"/> containing all the data needed
+        /// for processing this conversation turn. </param>
+        /// <param name="cancellationToken">(Optional) A <see cref="CancellationToken"/> that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A <see cref="Task"/> that represents the work queued to execute.</returns>
+        public async Task OnTurnAsync(ITurnContext context, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // See https://aka.ms/about-bot-activity-message to learn more about message and other activity types.
+            switch (context.Activity.Type)
+            {
+                case ActivityTypes.Message:
+                    // Process on turn input (card or NLP) and gather new properties
+                    // OnTurnProperty object has processed information from the input message activity.
+                    var onTurnProperties = await DetectIntentAndEntitiesAsync(context);
+                    if (onTurnProperties == null)
+                    {
+                        break;
+                    }
+
+                    // Set the state with gathered properties (intent/ entities) through the onTurnAccessor
+                    await _onTurnAccessor.SetAsync(context, onTurnProperties);
+
+                    // Create dialog context.
+                    var dc = await _dialogs.CreateContextAsync(context);
+
+                    // Continue outstanding dialogs.
+                    await dc.ContinueDialogAsync();
+
+                    // Begin main dialog if no outstanding dialogs/ no one responded
+                    if (!dc.Context.Responded)
+                    {
+                        await dc.BeginDialogAsync(nameof(MainDispatcher));
+                    }
+
+                    break;
+                case ActivityTypes.ConversationUpdate:
+                    // Welcome user.
+                    await WelcomeUserAsync(context);
+                    break;
+                default:
+                    // Handle other activity types as needed.
```

Comment:
>// Handle other activity types as needed [](start = 20, length = 40)

await turnContext.SendActivityAsync($"{turnContext.Activity.Type} event detected");

add this here